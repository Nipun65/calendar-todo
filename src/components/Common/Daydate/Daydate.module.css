.days {
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-template-rows: repeat(1, 1fr);
  padding: 15px 0;
  font-size: 1.5rem;
  font-weight: 400;
  letter-spacing: 0.1rem;
}

.days-ele {
  display: flex;
  justify-content: center;
}

.array-element {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 85px;
}

.array {
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-template-rows: repeat(7, 1fr);
  height: 85px;
}

.array-year {
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-template-rows: repeat(7, 1fr);
  height: 50px;
}

.parent-element:hover {
  background-color: rgb(60, 64, 67, 0.1);
}

.array-element-year {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 50px;
}

.month {
  font-size: 1.2rem;
  font-weight: 800;
  font-style: italic;
  border: none;
  border-bottom: 1px dotted black;
  margin: 20px;
  padding: 10px;
}

.grid-item {
  border: 1px solid black;
  padding: 18px 18px;
  text-align: center;
}

@media only screen and (max-width: 1200px) {
  .days {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    grid-template-rows: repeat(1, 1fr);
    font-size: 1.1rem;
  }
}

@media only screen and (max-width: 900px) {
  .days {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    grid-template-rows: repeat(1, 1fr);
    font-size: 1rem;
  }
}

@media only screen and (max-width: 600px) {
  .days {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    grid-template-rows: repeat(1, 1fr);
    font-size: 0.9rem;
  }
}
