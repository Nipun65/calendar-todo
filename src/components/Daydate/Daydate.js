import styles from "./Daydate.module.css";

const Daydate = (props) => {
  const days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
  const startYear = props.date.year;
  const startMonth = props.date.month;

  let daywisedates = Array(6)
    .fill()
    .map(() => Array(7).fill(0));

  const fillTheArray = (i, startDay, end) => {
    let date = 1;
    let week = 0;
    while (date <= end) {
      daywisedates[week][startDay % 7] = date;
      startDay++;
      date++;
      if (startDay % 7 === 0) {
        week++;
      }
    }
  };

  const startDate = new Date(startYear, startMonth, 1);
  const endDate = new Date(startYear, startMonth + 1, 0);
  const endDateOfTheMonth = endDate.getDate();
  const startDayOfTheMonth = startDate.getDay();
  fillTheArray(startMonth, startDayOfTheMonth, endDateOfTheMonth);

  const dateHandler = (event) => {
    props.date.selectedDate = {
      userYear: props.date.year,
      userMonth: props.date.month,
      userDate: +event.target.textContent,
    };
    props.setValue(props.date);
  };
  return (
    <div
      className={`${styles["main-div"]} ${
        props.selectedOption === "Year" ? styles["grid-item"] : ""
      }`}
    >
      {props.selectedOption === "Year" && (
        <div className={styles["month"]}>{props.month[props.date.month]}</div>
      )}
      <div className={styles["days"]}>
        {days.map((value) => (
          <div key={Math.random()} className={styles["days-ele"]}>
            {value}
          </div>
        ))}
      </div>
      <div>
        {daywisedates.map((value) => (
          <div
            className={`${
              props.selectedOption === "Year"
                ? styles["array-year"]
                : styles["array"]
            }`}
            key={Math.random()}
          >
            {value.map((date, index) => {
              if (date) {
                return (
                  <div
                    key={Math.random()}
                    className={`${
                      props.selectedOption === "Year"
                        ? styles["array-element-year"]
                        : styles["array-element"]
                    }`}
                    onClick={dateHandler}
                    style={
                      props.date.month === props.date.currentMonth &&
                      props.date.date === date &&
                      props.date.currentYear === +props.date.year
                        ? { backgroundColor: "#81d5ea" }
                        : props.date.selectedDate.userDate === date &&
                          props.date.month === props.date.selectedDate.userMonth
                        ? { backgroundColor: "rgb(201,234,255)", opacity: 0.5 }
                        : {}
                    }
                  >
                    {date}
                  </div>
                );
              } else {
                return (
                  <div
                    key={Math.random()}
                    className={styles["array-element"]}
                  ></div>
                );
              }
            })}
          </div>
        ))}
      </div>
    </div>
  );
};
export default Daydate;
