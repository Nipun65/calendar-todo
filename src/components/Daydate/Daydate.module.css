.days {
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-template-rows: repeat(1, 1fr);
  padding-bottom: 1.1em;
  font-size: 1.5rem;
  font-weight: 400;
  letter-spacing: 0.1rem;
  text-shadow: 0 0.3rem 0.5rem rgba(0, 0, 0, 0.5);
}
.days-ele {
  display: flex;
  justify-content: center;
}

.array-element {
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  height: 85px;
}
.array-element :hover {
  background-color: azure;
}

.array {
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-template-rows: repeat(7, 1fr);
  height: 85px;
}

.array-year {
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-template-rows: repeat(7, 1fr);
  height: 50px;
}
.array-element-year {
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  height: 50px;
}

.month {
  font-size: 1.2rem;
  font-weight: 800;
  /* padding: 20px 20px; */
  font-style: italic;
  border: none;
  border-bottom: 1px dotted black;
  margin: 20px;
  padding: 10px;
  /* text-decoration: dotted; */
}

.main-div {
  height: fit-content;
}

.grid-item {
  border: 1px solid black;
  padding: 18px 18px;
  text-align: center;
}

@media only screen and (max-width: 600px) {
  .days {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    grid-template-rows: repeat(1, 1fr);
    padding-bottom: 1.1em;
    font-size: 0.1rem;
    font-weight: 50;
    letter-spacing: 0.1rem;
    text-shadow: 0 0.3rem 0.5rem rgba(0, 0, 0, 0.5);
  }
  .days-ele {
    display: flex;
    justify-content: center;
    padding: 4px 4px;
  }
}

@media only screen and (max-width: 900px) {
  .days {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    grid-template-rows: repeat(1, 1fr);
    padding-bottom: 1.1em;
    font-size: 0.1rem;
    font-weight: 50;
  }
  .days-ele {
    display: flex;
    justify-content: center;
    padding: 4px 4px;
  }
}

@media only screen and (max-width: 1200px) {
  .days {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    grid-template-rows: repeat(1, 1fr);
    padding-bottom: 1.1em;
    font-size: 1.1rem;
    font-weight: 200;
  }
}
