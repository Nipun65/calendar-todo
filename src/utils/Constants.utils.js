export const MONTHS = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

export const DROPDOWNLIST = ['Month', 'Year'];

export const DAYS = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];

export const DATE = {
  year: 2023,
  month: 0,
  date: 31,
  currentMonth: 0,
  currentYear: 2023,
  selectedDate: { userYear: 2023, userMonth: 0, userDate: 31 },
};
