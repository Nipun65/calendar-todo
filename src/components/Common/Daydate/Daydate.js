import styles from './Daydate.module.css';
import { DAYS } from '../../../utils/Constants.utils';

function Daydate({ date, setValue, selectedOption, MONTHS, setView }) {
  const startYear = date.year;
  const startMonth = date.month;
  const daywisedates = Array(6)
    .fill()
    .map(() => Array(7).fill(0));

  // filling the array according to startday and end date
  const fillTheArray = (i, startDay, end) => {
    let dateValue = 1;
    let week = 0;
    while (dateValue <= end) {
      daywisedates[week][startDay % 7] = dateValue;
      startDay++;
      dateValue++;
      if (startDay % 7 === 0) {
        week++;
      }
    }
  };

  const startDate = new Date(startYear, startMonth, 1);
  const endDate = new Date(startYear, startMonth + 1, 0);
  const endDateOfTheMonth = endDate.getDate();
  const startDayOfTheMonth = startDate.getDay();
  fillTheArray(startMonth, startDayOfTheMonth, endDateOfTheMonth);

  const dateHandler = (event) => {
    date.selectedDate = {
      userYear: date.year,
      userMonth: date.month,
      userDate: +event.target.textContent,
    };
    setValue(date);
    setView('day');
  };

  // style for setting today's date and user selected date
  const getStyle = (dateValue) => {
    if (
      date.month === date.currentMonth &&
      date.date === dateValue &&
      date.currentYear === +date.year
    ) {
      return { backgroundColor: '#95638e4d' };
    }
    return {};
  };

  return (
    <div
      className={`${styles['main-div']} ${
        selectedOption === 'Year' ? styles['grid-item'] : ''
      }`}
      data-testid="daydate"
    >
      {selectedOption === 'Year' && (
        <div className={styles.month}>{MONTHS[date.month]}</div>
      )}
      <div className={styles.days}>
        {DAYS.map((value) => (
          <div key={Math.random()} className={`${styles['days-ele']}`}>
            {value}
          </div>
        ))}
      </div>
      <div>
        {daywisedates.map((value) => (
          <div
            key={Math.random()}
            className={`${
              selectedOption === 'Year' ? styles['array-year'] : styles.array
            }`}
          >
            {value.map((dateValue) => {
              if (dateValue) {
                return (
                  <div
                    className={`${styles['parent-element']}`}
                    key={Math.random()}
                  >
                    <div
                      role="presentation"
                      className={`${
                        selectedOption === 'Year'
                          ? styles['array-element-year']
                          : styles['array-element']
                      } is-clickable ${
                        dateValue !== 0 ? styles['hover-element'] : ''
                      }`}
                      onClick={dateHandler}
                      style={getStyle(dateValue)}
                    >
                      {dateValue}
                    </div>
                  </div>
                );
              }
              return (
                <div
                  key={Math.random()}
                  className={`${
                    selectedOption === 'Year'
                      ? styles['array-element-year']
                      : styles['array-element']
                  } `}
                />
              );
            })}
          </div>
        ))}
      </div>
    </div>
  );
}

export default Daydate;
