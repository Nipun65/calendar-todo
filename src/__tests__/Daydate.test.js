import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
import Daydate from '../components/Common/Daydate/Daydate';
import '@testing-library/jest-dom/extend-expect';
import { DATE } from '../utils/Constants.utils';

afterEach(() => {
  cleanup();
});

it('should match the snapshot', () => {
  const tree = renderer.create(<Daydate date={DATE} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render Card Component', () => {
  render(<Daydate date={DATE} />);
  const mainDiv = screen.getByTestId('daydate');
  expect(mainDiv).toBeInTheDocument();

  const suntxt = screen.queryByText('Sun');
  expect(suntxt).toBeInTheDocument();
  const Montxt = screen.queryByText('Mon');
  expect(Montxt).toBeInTheDocument();
  const Tuetxt = screen.queryByText('Tue');
  expect(Tuetxt).toBeInTheDocument();
  const Wedtxt = screen.queryByText('Wed');
  expect(Wedtxt).toBeInTheDocument();
  const Thutxt = screen.queryByText('Thu');
  expect(Thutxt).toBeInTheDocument();
  const Fritxt = screen.queryByText('Fri');
  expect(Fritxt).toBeInTheDocument();
  const Sattxt = screen.queryByText('Sat');
  expect(Sattxt).toBeInTheDocument();

  const secondChild = mainDiv.children[1];
  const childCount = secondChild.children.length;
  expect(childCount).toBe(6);
});
