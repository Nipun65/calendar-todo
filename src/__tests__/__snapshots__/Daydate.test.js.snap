// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="main-div "
  data-testid="daydate"
>
  <div
    className="days"
  >
    <div
      className="days-ele"
    >
      Sun
    </div>
    <div
      className="days-ele"
    >
      Mon
    </div>
    <div
      className="days-ele"
    >
      Tue
    </div>
    <div
      className="days-ele"
    >
      Wed
    </div>
    <div
      className="days-ele"
    >
      Thu
    </div>
    <div
      className="days-ele"
    >
      Fri
    </div>
    <div
      className="days-ele"
    >
      Sat
    </div>
  </div>
  <div>
    <div
      className="array"
    >
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          1
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          2
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          3
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          4
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          5
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          6
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          7
        </div>
      </div>
    </div>
    <div
      className="array"
    >
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          8
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          9
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          10
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          11
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          12
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          13
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          14
        </div>
      </div>
    </div>
    <div
      className="array"
    >
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          15
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          16
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          17
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          18
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          19
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          20
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          21
        </div>
      </div>
    </div>
    <div
      className="array"
    >
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          22
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          23
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          24
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          25
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          26
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          27
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          28
        </div>
      </div>
    </div>
    <div
      className="array"
    >
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          29
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={Object {}}
        >
          30
        </div>
      </div>
      <div
        className="parent-element"
      >
        <div
          className="array-element is-clickable hover-element"
          onClick={[Function]}
          role="presentation"
          style={
            Object {
              "backgroundColor": "#4c7bde5c",
            }
          }
        >
          31
        </div>
      </div>
      <div
        className="array-element "
      />
      <div
        className="array-element "
      />
      <div
        className="array-element "
      />
      <div
        className="array-element "
      />
    </div>
    <div
      className="array"
    >
      <div
        className="array-element "
      />
      <div
        className="array-element "
      />
      <div
        className="array-element "
      />
      <div
        className="array-element "
      />
      <div
        className="array-element "
      />
      <div
        className="array-element "
      />
      <div
        className="array-element "
      />
    </div>
  </div>
</div>
`;
